### PR TITLE
create dictionary with values as squares for their keys

### PR DESCRIPTION
**Pull request by JanetJanx**
Hello @dennisja, I have modified my dictionary function and also added tests for it in (dictionary_test).

### CODE REVIEW

**Comment by JanetJanx** [comment]
Hi @dennisja, you didn't look at this

File: dictionary_test.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import unittest
+
+from dictionary import dictinary
+
+class DictTest(unittest.TestCase):
+
+    def test_dictinary(self):
+        self.assertEqual(dictinary(3,8), {3: 9, 4: 16, 5: 25, 6: 36, 7: 49})
```
Comment (dennisja) [inline dictionary_test.py]: Write a test that loops through the dictionary and asserts whether each value is the square of each key.
Hint:
```py
for key, value in generated_dictionary:
    # do the assertion here
```

**Review by dennisja — CHANGES_REQUESTED** [review]

File: dictionary.py [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 #function to generate the dictionary keys and values
-def dictinary():
-    #create dictionary keys as numbers with values as their squares
-    for n in range(1,16):
-        sq_number[n] = n**2
-    return sq_number
+def dictinary(a,b):
+    if type(a) == int and type(b) == int:
+        #variable for the empty dictionary
+        sq_number = dict()
```
Comment (dennisja) [inline dictionary.py]: Would appreciate if you can reduce lines 5-9 into comprehension. You can read about comprehension [here](https://medium.com/the-andela-way/a-cheap-introduction-to-comprehension-in-python-2269895f996f)

Comment (JanetJanx) [reply dictionary.py]: @dennisja , I read through the concept and it's really good. I have manged to use it in the recent changes.

File: dictionary_test.py (hunk 1 above)
Comment (JanetJanx) [reply dictionary_test.py]: Hi @dennisja , I have tried to use a loop to test whether each value is a square of each key.
You may look through the recent changes please

**Comment by JanetJanx** [comment]
Hi @dennisja , I made the changes, please look through them.

File: dictionary_test.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+from dictionary import dictinary
+
+class DictTest(unittest.TestCase):
+
+    def test_dictinary(self):
+        generated_dictionary = dictinary(3,8)
+        """test whether each value is the square of each key."""
+        for key, value in generated_dictionary.iteritems():
+            self.assertEquals(generated_dictionary[key], key**2)
```
Comment (dennisja) [inline dictionary_test.py]: Use assertEqual instead

Comment (dennisja) [inline dictionary_test.py]: Check whether value is a square of the key instead

File: dictionary_test.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import unittest
+
+from dictionary import dictinary
+
+class DictTest(unittest.TestCase):
+
+    def test_dictinary(self):
+        generated_dictionary = dictinary(3,8)
+        """test whether each value is the square of each key."""
+        for key, value in generated_dictionary.iteritems():
```
Comment (dennisja) [inline dictionary_test.py]: use items() in place of iteritems

File: dictionary.py [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
+        #lets create a dictionary with keys as numbers with values as their squares
+        sq_number = {n:n**2 for n in range(a,b)}
+        return sq_number
+    else:
+        return "Please use numbers(integers)"
 
 #print the generated dictinary
-print dictinary()
\ No newline at end of file
+print dictinary(4,15)
```
Comment (dennisja) [inline dictionary.py]: Use the print builtin function

**Review by dennisja — CHANGES_REQUESTED** [review]

**Pull request merged**